docs/CI-CD.md

```diff
@@ -0,0 +1,93 @@
+# EVE CI/CD infrastructure
+
+EVE development happens on GitHub and we use GitHub's Actions as the core of our CI/CD infrastructure.
+Within GitHub actions we try to do as much as possible with the stock `ubuntu-20.04` build VM that
+gets instantiated from scratch every time jobs need to run as part of a given action. However, since
+we also have to produce ARM builds, we have a hosted runner tagged `ARM64` that we use for subset of
+the actions. EVE uses a standard OCI Registry (currently on [Docker HUB](https://hub.docker.com/r/lfedge/eve))
+to store all of its CI/CD build artifacts.
+
+Definitions of EVE CI/CD actions are available under [.github/workflows](../.github/workflows) folder
+and they currently include:
+
+* [Publish](https://github.com/lf-edge/eve/actions?query=workflow%3APublish) publishes various EVE build artifacts into an OCI Registry
+* [Pulls](https://github.com/lf-edge/eve/actions?query=workflow%3APRs) manages pull requests
+* [Apache Yetus](https://github.com/lf-edge/eve/actions?query=workflow%3A%22Apache+Yetus%22) runs static, lint-like [Apache Yetus](http://yetus.apache.org/) checks on EVE
+* [Eden](https://github.com/lf-edge/eve/actions?query=workflow%3AEden) runs integration tests from EVE's sister project [Eden](https://github.com/lf-edge/eden)
+
+Since EVE community generally frowns on branch-heavy development most of the descriptions below
+will only apply to the master branch (unless stated otherwise).
+
+## Publish
+
+This is the only action that uses both x86 and ARM64 runners. It gets triggered on either updates
+to the `master` branch or creation of release tags. Publish produces a set of tagged artifacts and
+pushes them into EVE's OCI Registry containers. Logic responsible for figuring out correct tags
+for these artifacts resides in EVE's top-level Makefile and can be summarized as:
+
+* If Publish gets triggered on a release tag (release tags are the ones with `x.y.z` names) that
+  compares as grater than all the other release tags currently present in a repo that release tag
+  is considered to be the latest. All artifacts originating from the Publish triggered on the latest
+  tag get a matching `x.y.z` OCI registry tags AND get an additional symbolic tag `latest`. Since
+  the artifacts are typically retrieved with a `docker pull lfedge/eve-XXX` this policy corresponds
+  well with the expectation that such a pull is expected to retrieve `lfedge/eve-XXX:latest` even
+  though it is a mere alias for `lfedge/eve-XXX:x.y.z`. Another crucial distinction of a Publish
+  action triggered on the latest tag is that it gets the benefit of a full (from scratch) build
+  as opposed to using pre-cached artifacts. This is done to ensure that EVE builds can be rebuilt
+  in the future in a self-contained manner but it does, of course, increase the build time dramatically.
```

Comment:
I don't understand this. When we build on tags, we _rebuild_ as opposed to just tagging the existing images, a la:

```
docker tag lfedge/eve-foo:abcd555 lfedge/eve-foo:v1.2.3
```

Why not, if the hash (in the above example `abcd555`) is identical?

Reply:
Because we actually forcing that full rebuild (the trick we play is passing `--hash x.y.z --release latest` combo to linuxkit). This is very much on purpose as a forcing function for us to keep fixing weird Alpine issues in our Dockerfiles.